oaknational/Oak-Web-Application
# feat: pupil expiring banners

### Job Story

**When** I navigate to a lesson or lesson listing which is due to be expired
**I want to see the new lesson takedown banner**
**So that users can plan appropriately**.

### Implementation details

The banner appears in response to the action “display_expiring_banner”

The current expiring banner appears here:
https://www.thenational.academy/pupils/programmes/maths-secondary-year-10-core-l/units/trigonometry-1-0125/lessons/applying-trigonometry-6mup2c/overview

Design docs:
https://www.figma.com/design/dGO9erL0b0kpYxmv6Jku4f/Lessons-takedown-banners?node-id=2011-7642&m=dev


## How to test

1. Go to https://deploy-preview-3047--oak-web-application.netlify.thenational.academy/pupils/programmes/maths-secondary-year-10-core-l/units/trigonometry-1-0125/lessons/applying-trigonometry-6mup2c/overview
2. you should see the expiring banner
3. click the button. It should take you to the programme options page

Further testing:
Unfortunately we do not currently have any other expired legacy content for testing in production.

### Acceptance criteria

- [x]  banner matches design
- [x]  link goes to appropriate place
- [x]  pupil lesson page
- [x]  pupil lesson canonical page
- [x]  pupil lesson listing
